### PR TITLE
Add AdminOrgVdcStorageProfile for querying (improved #375)

### LINKED ISSUE #373: Unmarshal of QueryResultRecordsType for OrgVdcStorageProfileRecord
Hi!

On struct QueryResultRecordsType you have element OrgVdcStorageProfileRecord with same xml unmarshal tag. 
https://github.com/vmware/go-vcloud-director/blob/v2.11.0/types/v56/types.go#L2137
```go
type QueryResultRecordsType struct {
...
	OrgVdcStorageProfileRecord      []*QueryResultOrgVdcStorageProfileRecordType      `xml:"OrgVdcStorageProfileRecord"`      // A record representing storage profiles

```

But vCloud returns xml element AdminOrgVdcStorageProfileRecord with same content.

In result - i have `1` qty in total with empty OrgVdcStorageProfileRecord

```go
results, err := vCloud.cli[vcloudID].QueryWithNotEncodedParams(nil, map[string]string{
	"type":          "adminOrgVdcStorageProfile",
	"filter":        fmt.Sprintf("id==%s", url.QueryEscape(avdc.AdminVdc.VdcStorageProfiles.VdcStorageProfile[0].ID)),
	"filterEncoded": "true",
})
``` 

I want to send PR, but i don't know what you want to do:
- change unmarshal tag
- add another Element in struct with name AdminOrgVdcStorageProfileRecord

And i want to add:
- type for query `adminOrgVdcStorageProfile`
- func for querying StorageProfile of ovdc by urn(id)

### PR DESCRIPTION
**Pull request by Didainius**
This PR builds on top of discussions in #375 with test improvement.

Closes #373, #375

### CODE REVIEW

**Review by lvirbalas — APPROVED** [review]
Thank you!

File: govcd/system.go [hunk 1]
```diff
@@ -1115,3 +1115,47 @@ func (vcdClient *VCDClient) GetOrgList() (*types.OrgList, error) {
 	}
 	return orgList, nil
 }
+
+// QueryAdminOrgVdcStorageProfileByID finds a StorageProfile of VDC by ID as admin
+func QueryAdminOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) (*types.QueryResultAdminOrgVdcStorageProfileRecordType, error) {
+	if !vcdCli.Client.IsSysAdmin {
+		return nil, errors.New("cant query type QueryAdminOrgVdcStorageProfileByID as tenant user")
```
Comment (adambarreiro) [inline govcd/system.go:1122]: Nitpick:
```suggestion
		return nil, errors.New("can't query type QueryAdminOrgVdcStorageProfileByID as tenant user")
```

File: govcd/system.go [hunk 2]
```diff
@@ -1115,3 +1115,47 @@ func (vcdClient *VCDClient) GetOrgList() (*types.OrgList, error) {
...
+	if len(results.Results.AdminOrgVdcStorageProfileRecord) > 1 {
+		return nil, fmt.Errorf("more than one Storage Profile found with ID %s", id)
+	}
+	return results.Results.AdminOrgVdcStorageProfileRecord[0], nil
+}
+
+// QueryOrgVdcStorageProfileByID finds a StorageProfile of VDC by ID
+func QueryOrgVdcStorageProfileByID(vcdCli *VCDClient, id string) (*types.QueryResultOrgVdcStorageProfileRecordType, error) {
+	if vcdCli.Client.IsSysAdmin {
+		return nil, errors.New("cant query type QueryOrgVdcStorageProfileByID as sys admin")
```
Comment (adambarreiro) [inline govcd/system.go:1144]: Nitpick:
```suggestion
		return nil, errors.New("can't query type QueryOrgVdcStorageProfileByID as System administrator")
```

File: types/v56/types.go [hunk 3]
```diff
@@ -2192,6 +2192,7 @@ type QueryResultRecordsType struct {
 	VAppRecord                      []*QueryResultVAppRecordType                      `xml:"VAppRecord"`                      // A record representing a VApp result.
 	AdminVAppRecord                 []*QueryResultVAppRecordType                      `xml:"AdminVAppRecord"`                 // A record representing a VApp result as admin.
 	OrgVdcStorageProfileRecord      []*QueryResultOrgVdcStorageProfileRecordType      `xml:"OrgVdcStorageProfileRecord"`      // A record representing storage profiles
+	AdminOrgVdcStorageProfileRecord []*QueryResultAdminOrgVdcStorageProfileRecordType `xml:"AdminOrgVdcStorageProfileRecord"` // A record Admin representing storage profiles
```
Comment (adambarreiro) [inline types/v56/types.go:2195]: Nitpick
```suggestion
	AdminOrgVdcStorageProfileRecord []*QueryResultAdminOrgVdcStorageProfileRecordType `xml:"AdminOrgVdcStorageProfileRecord"` // A record representing storage profiles as admin
```

**Review by adambarreiro — APPROVED** [review]
Just a few nitpicks 😊

File: govcd/system.go (hunk 1 above)
Comment (Didainius) [reply govcd/system.go]: Addressed.

File: govcd/system.go (hunk 2 above)
Comment (Didainius) [reply govcd/system.go]: Addressed.

File: types/v56/types.go (hunk 3 above)
Comment (Didainius) [reply types/v56/types.go]: Addressed.

**Pull request merged**